/*
 * Copyright 2024 Sowers, LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
syntax = "proto3";

option go_package = "bosca.io/api/protobuf/bosca/content";

import "bosca/content/model.proto";

package bosca.content;

enum StorageSystemType {
  unknown_storage_system = 0;
  vector_storage_system = 1;
  search_storage_system = 2;
  metadata_storage_system = 3;
  supplementary_storage_system = 4;
}

message StorageSystem {
  string id = 1;
  StorageSystemType type = 2;
  string name = 3;
  string description = 4;
  map<string, string> configuration = 5;
}

message StorageSystems {
  repeated StorageSystem systems = 1;
}

message StorageSystemModel {
    Model model = 2;
    string type = 3;
}

message StorageSystemModels {
    repeated StorageSystemModel models = 1;
}